core-bundle/src/EventListener/MergeHttpHeadersListener.php

```diff
@@ -144,6 +144,11 @@ private function setResponseHeaders(Response $response)
         $allowOverrides = [];
 
         foreach ($this->headers as $header) {
+            if (preg_match('/^HTTP\/[^ ]* (\d{3}) (.*)$/i', $header, $matches)) {
```

Comment:
Not sure about this regular expression. Imho there must be something after `HTTP/` so it should not be `*` and imho after the status code there must be no message so matching the ` ` is not correct either.
I would be defensive here and use `/^HTTP\/[^ ]+ (\d{3})( .*)?$/i` and only set the message (trim it) when present.

Reply:
If I understand the RFC correctly, there's always a _Reason Phrase_
https://tools.ietf.org/html/rfc2616#section-6.1

Reply:
Actually both, HTTP version and reason phrase are not required anymore since HTTP2 https://http2.github.io/http2-spec/#rfc.section.8.1.2.4
Also see https://github.com/http2/http2-spec/issues/202

Just saying, it is possible although unlikely :)

Reply:
well in that case the Header wouldn't match the regex, which seems fine because we can't adjust the response anyway? We might only need an additional check if there is a colon at all?

Reply:
Why? We sure can. Just leave out the message, it's optional in `setStatusCode()` as well.

Reply:
improved in ab5ae323c0d6a3a937376e070c6a7588bf9dcc5a